.buildkite/pipeline.yml

```diff
@@ -136,7 +145,10 @@ steps:
       distro: "${BUILDKITE_AGENT_META_DATA_DISTRO}"
       hostname: "${BUILDKITE_AGENT_META_DATA_HOSTNAME}"
 
-  - wait
+  # This makes sure next step will run even if previous steps failed
+  # to make sure we can reach the final step for resource cleanup 
+  - wait: ~
+    continue_on_failure: true
```

Comment:
This is needed to guarantee we always able to reach to final stage for cleanup. 

Previously, if we have failures in early stages, the build will stop. Multiple failures without resource cleanup will eat up the storage resulting [No space left error](https://buildkite.com/firecracker-microvm/firecracker-go-sdk/builds/3219#0189429f-e175-48cf-8bdc-10e0775ebf1a).

Reply:
I would suggest decoupling the pipeline changes from the dependency upgrade. Split it into two commits or two PRs

Reply:
I splitted into two  commits. 
PTAL @swagatbora90